digdag-core/src/main/java/io/digdag/core/agent/MultiThreadAgent.java

```diff
@@ -98,12 +118,18 @@ public void run()
                                     logger.error("Uncaught exception. Task queue will detect this failure and this task will be retried later.", t);
                                     errorReporter.reportUncaughtError(t);
                                 }
+                                finally {
+                                    synchronized (taskCountLock) {
+                                        activeTaskCount--;
```

Comment:
`SynchronousQueue` is used with `maximumPoolSize: Integer.MAX_VALUE`. So it's usually okay. But it seems a bit naive to me...